### ISSUE TITLE
confusions about the particle Calorimeter plugin

### ISSUE THREAD

**Issue by zwjlpi** [ComputationalRadiationPhysics/picongpu#4946]
The particle calorimeter plugin was supposed to store the particles that leave the simulation box in the spatial-energy matrix. But when I add the codes in the example .cfg file and collect the particles at full solid angle, the energy spectrum obtained from the calorimeter seems almost the same with the histogram plugin result. Does the calorimeter plugin only reserved the data of particles that reach the absorbing boundary region at this output step interval, rather than all left particles from the begin of simulation to this instant? If not, how can I activate the function to record the data of all left or leaving particles?

**Comment by psychocoderHPC** [comment]
Yes, the calorimeter plugin is only collecting particles those leafing the global domain on absorbing particle boundaries together with all particles within the domain.
Monitoring only leaving particles is not possible without code changes.
Particles copied from one domain to another will can not be monitored with the calocimeter.

We have a new generic binning plugin. https://picongpu.readthedocs.io/en/latest/usage/plugins/binningPlugin.html#usage-plugins-binningplugin

example:

https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/include/picongpu/param/binningSetup.param

In prinziple you can define any kind of filter to select particles within a region only. Currently it is not possible to describe very easy that only particles those passing a slice into one direction should be taken into account.

@ikbuibui We should provide the possibility to the flexible binning plugin that we can bin leaving particles only. There is a method in our plugin interface `onParticleLeave` where all plugins get notified if they leave the global domain.

**Comment by ikbuibui** [comment]
👍 I will prioritize this over the other todos on the binning roadmap

**Comment by zwjlpi** [comment]
@psychocoderHPC Thank you so much for these helpful reply and suggestions. I'm trying to use the filters that attach near the boundarys to extract the particles and dump them every simulation time step. According to the variation on id data between two neighboring outputs, the particles  left the box at each step could be determined. This is not convenient but practical. The binning plugin is currently to complex for me, and we all look forward the formal binning plugin in the upcoming new picongpu release.

**Comment by steindev** [comment]
@zwjlpi Do I get it right that your issue is resolved for now? If so please close the issue. Otherwise you can post another question or open a new issue

**Comment by psychocoderHPC** [comment]
@zwjlpi @ikbuibui opened a PR to create histograms out of particles leaving the outer simulation box https://github.com/ComputationalRadiationPhysics/picongpu/pull/5039

It is not collecting particles on periodic boundaries or reflective boundaries.
